python/src/aac/io/paths.py

```diff
@@ -0,0 +1,57 @@
+"""Module for shared path functions."""
+
+import logging
+import os
+import unicodedata
+from urllib.parse import unquote
+
+
+def sanitize_filesystem_path(file_path: str) -> str:
+    r"""
```

Comment:
It's a flake rule. Removing the regular expression leads flake8 to report the following findings:
```
./src/aac/io/paths.py:10:1: D301 Use r""" if any backslashes in a docstring
    """
    Sanitizes user-provided file path strings to prevent injection and tampering of filesystem paths.

    This function does not determine if a user can write to the path location, it only sanitizes and evaluates the
    path to a full path. The user must still check that the path is a valid destination.

    This function performs the following sanitizing actions:

        * Removes any URL encoding
        * Converts the path to an ASCII encoding
        * Replaces instances of / or \ with os-specific path separators
        * Normalizes the path
        * Determines the realpath if symbolic links are used
        * Ensures that the path is an absolute path

    Path Traversal Security References:

        * OWASP https://owasp.org/www-community/attacks/Path_Traversal

    Args:
        file_path (str): The file path to sanitize

    Returns:
        A sanitized filepath.
    """
^
./src/aac/io/paths.py:20:38: W605 invalid escape sequence '\ '
    """
    Sanitizes user-provided file path strings to prevent injection and tampering of filesystem paths.

    This function does not determine if a user can write to the path location, it only sanitizes and evaluates the
    path to a full path. The user must still check that the path is a valid destination.

    This function performs the following sanitizing actions:

        * Removes any URL encoding
        * Converts the path to an ASCII encoding
        * Replaces instances of / or \ with os-specific path separators
        * Normalizes the path
        * Determines the realpath if symbolic links are used
        * Ensures that the path is an absolute path

    Path Traversal Security References:

        * OWASP https://owasp.org/www-community/attacks/Path_Traversal

    Args:
        file_path (str): The file path to sanitize

    Returns:
        A sanitized filepath.
    """
       
                             ^
1     D301 Use r""" if any backslashes in a docstring
1     W605 invalid escape sequence '\ '
2
```

Reply:
Oh, interesting. It's not a huge deal now that I know but it might be better to say "normalize path separators" or something so we can avoid that.